gh-135906: Use _PyObject_CAST in internal CPython headers (#135892)
python/cpython: Misc/NEWS.d/next/C_API/2025-06-25-01-03-10.gh-issue-135906.UBrCWq.rst

> Fix compilation errors when compiling the internal headers with a C++ compiler.

Reviewer: compiler errors or compiler warnings?

Author: Compilation error

Reviewer: But is it because you treat warnings as errors in your CI?

Reviewer: `-Werror` is not used on the CI. So there are really compiler errors, ok ;-)

> Fix compilation errors when compiling with a C++ compiler and internal CPython headers are (indirectly) included

Reviewer 2: We can make this a little less wordy:
```suggestion
Fix compilation errors when compiling the internal headers with a C++ compiler.
```
